source/write-operations/bulk-writes.txt

```diff
@@ -1,58 +1,347 @@
+   :end-before: end-bulk-replace-one
+   :language: java
+   :copyable:
+   :dedent:
+
+To replace multiple documents, you must create an instance of ``ReplaceOneModel`` for each document.
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To delete a document, create an instance of ``DeleteOneModel`` and pass in a
+query filter specifying the document you want to delete. ``DeleteOneModel`` removes
+only *the first* document that matches your query filter.
+
+The following example creates an instance of ``DeleteOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.java
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: java
+   :copyable:
+   :dedent:
+
+To delete multiple documents, create an instance of ``DeleteManyModel`` and pass in a
+query filter specifying the document you want to delete. ``DeleteManyModel`` removes
+*all* documents that match your query filter.
+
+The following example creates an instance of ``DeleteManyModel``:
+
+.. literalinclude:: /includes/write/bulk-write.java
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: java
+   :copyable:
+   :dedent:
+
+Call the ``bulkWrite()`` Method
+--------------------------------
+
+After you define a class instance for each operation you want to perform,
+pass a list of these instances to the ``bulkWrite()`` method.
+By default, the method runs the operations in the order
+they're defined in the list.
+
+The following example performs multiple write operations by using the
+``bulkWrite()`` method:
+
+.. io-code-block::
+
+   .. input:: /includes/write/bulk-write.java
+      :start-after: start-bulk-write-mixed
+      :end-before: end-bulk-write-mixed
+      :language: java
+      :copyable:
+      :dedent:
+
+   .. output::
+      
+      AcknowledgedBulkWriteResult{insertedCount=2, matchedCount=2, removedCount=1, modifiedCount=2, upserts=[], inserts=[BulkWriteInsert{index=0, id=BsonObjectId{value=66a3cebad6d5d5070c7883e3}}, BulkWriteInsert{index=1, id=BsonObjectId{value=66a3cebad6d5d5070c7883e4}}]}
+
+If any of the write operations fail, the {+driver-short+} raises a
+``BulkWriteError`` and does not perform any further operations.
+``BulkWriteError`` provides a ``details`` attribute that includes the operation
+that failed, and details about the exception.
+
+.. note::
+
+   When the {+driver-short+} runs a bulk operation, it uses the ``writeConcern`` of the
+   collection in which the operation is running. The driver reports all write
+   concern errors after attempting all operations, regardless of execution order. 
+
+Customize Bulk Write Operations
+-------------------------------
+
+The ``BulkWriteOptions`` class contains methods that modify
+the behavior of the ``bulkWrite()`` method. To use the ``BulkWriteOptions``
+class, construct a new instance of the class, then call one or more of its methods
+to modify the delete operation. You can chain these method calls together.
```

Comment:
I: why do these only modify delete operations? Or do all mentions of "delete operation" in this section need to be replaced with "bulk operation"?